### PR TITLE
Query history saving pref (#changed) and replace content fixes (#fixed)

### LINKED ISSUE #818: Query History saving multiple queries separated by semicolons as separate entries
- Sequel Ace Version (including build number): 3.1.1 Build 3016
- Sequel Ace Source (App Store/GitHub/Homebrew): App Store
- macOS Version: 11.1
- Processor Type (Intel/Apple): Apple M1
- MySQL Version: 5.7.17
- macOS Localization: US English

**Is Issue Present in [Latest Beta](https://github.com/Sequel-Ace/Sequel-Ace/releases)?**

**Description**
After updating to 3.1.1, Query History now saves multiple queries that were separated by semicolon as separate entries. Instead of showing multiple queries in the query window, there's only 1 line showing.

**Steps To Reproduce**
1. Run multiple queries in the query window, separated by semi colons and click Run All like this:
/* this is a test */
update mytable set name='hi';
select * from mytable;

2. Clear the query window
3. Call up Query History and select the last query of multiple queries but all the individual queries showed up as 1 entry. So the Query History shows up like this in the selection window:
select * from mytable;
update mytable set name='hi';
/* this is a test */

**Expected Behaviour**
Like prior to 3.1.1, the Query History should show multiple lines queries as 1 single entry and populate the Query Window as such.

### LINKED ISSUE #820: History/Favorites Replaces Editor Content
- Sequel Ace Version (including build number): 3.1.1 Build 3016
- Sequel Ace Source (App Store/GitHub/Homebrew): App Store 
- macOS Version: 11.1
- Processor Type (Intel/Apple): Apple M1
- MySQL Version: 5.7.17
- macOS Localization: US English

**Is Issue Present in [Latest Beta](https://github.com/Sequel-Ace/Sequel-Ace/releases)?**

**Description**
In Query Editor, these options don't work:
"History Replaces Editor Content"
and
"Favorites Replaces Editor Content"

**Steps To Reproduce**
1. With "History Replaces Editor Content" checked in Query Editor (1st icon)
2. Enter a query in the Query Editor
3. Highlight the query, select an item from Query History, the history item appends to the highlighted query and does not replace it

**Expected Behaviour**
In 3.) above, the selected history item should replace the highlighted query completely. Save issue with "Favorites Replaces Editor Content". 

**Related Issues**

**Additional Context**

### PR DESCRIPTION
**Pull request by jamesstout**
## Changes:
- Added pref to General pane called "Save query history individually". This is the default and stores each line as a history item in the db, delimited by `;\n`. If you switch it off, the entire query is stored as one item in the db.
- Fixed `History Replaces Editor Content` and `Favorites Replaces Editor Content` not working properly.

## Closes following issues:
- Closes #818 
- Closes #820  

## Tested:
- Processors:
  - [x] Intel
  - [ ] Apple Silicon
- macOS Versions:
  - [ ] 10.12.x (Sierra)
  - [ ] 10.13.x (High Sierra)
  - [ ] 10.14.x (Mojave)
  - [x] 10.15.x (Catalina)
  - [ ] 11.x (Big Sur)
- Xcode Version: 12.3 (12C33)
- Localizations:
  - [ ] English
  - [ ] Spanish
  - [ ] Other (please specify)
## Additional notes:

**TODO**: unions and subqueries...

### CODE REVIEW

File: Interfaces/Preferences.xib [hunk 1]
```diff
@@ -755,17 +766,19 @@ Gw
...
+                <constraint firstItem="947" firstAttribute="top" secondItem="957" secondAttribute="bottom" constant="2" id="HC4-0f-RPy"/>
                 <constraint firstItem="24" firstAttribute="top" secondItem="581" secondAttribute="bottom" constant="9" id="HLY-rd-SfS"/>
                 <constraint firstItem="947" firstAttribute="trailing" secondItem="569" secondAttribute="trailing" id="IYp-aC-TVp"/>
                 <constraint firstItem="F3U-SQ-W2d" firstAttribute="leading" secondItem="17" secondAttribute="leading" id="Im0-9V-68O"/>
                 <constraint firstItem="784" firstAttribute="trailing" secondItem="569" secondAttribute="trailing" id="JVt-FZ-AWU"/>
                 <constraint firstItem="785" firstAttribute="leading" secondItem="17" secondAttribute="leading" constant="20" id="JuX-m1-yDO"/>
                 <constraint firstItem="947" firstAttribute="leading" secondItem="569" secondAttribute="leading" id="KeG-OR-BDE"/>
+                <constraint firstItem="hMm-gH-Xw4" firstAttribute="top" secondItem="947" secondAttribute="bottom" constant="6" id="Liq-UM-y1F"/>
```
Comment (Jason-Morcos) [inline Interfaces/Preferences.xib:776]: Looks like all the other constraints (or most) are 9 or 2 values. I might just double check that the pattern for the new pref matches the patterns for the checkboxes in the other pref panes

**Review by Jason-Morcos — APPROVED** [review]

**Review by Jason-Morcos — APPROVED** [review]

**Comment by n9yty** [comment]
Can this get pushed out to a release?  The problem with using a Beta is that it can't access any stored configurations/etc.  It is a real pain to have no access to query history because everything is so mangled.

**Comment by Jason-Morcos** [comment]
> Can this get pushed out to a release?  The problem with using a Beta is that it can't access any stored configurations/etc.  It is a real pain to have no access to query history because everything is so mangled.

This is available in the latest release already. Note that the latest beta actually is lagging behind the latest official release - we've pushed a few rapid releases straight to release without beta.

**Comment by n9yty** [comment]
ARRGH.  Somehow mine got disconnected from the App store and wasn't updating any more.  No idea. Thanks!

**Pull request merged**